### PR TITLE
Fixed #24708 -- Handled non-string values in GenericIPAddressField.to_python()

### PR DESCRIPTION
**Pull request by Pradeek**
Handled a case of GenericIPAddressField raising an exception if non
strings are passed in.

### CODE REVIEW

File: django/db/models/fields/__init__.py [hunk 1]
```diff
@@ -1963,6 +1963,8 @@ def get_internal_type(self):
         return "GenericIPAddressField"
 
     def to_python(self, value):
+        if value and not isinstance(value, six.string_types):
+            value = smart_text(value)
```
Comment (timgraham) [inline django/db/models/fields/__init__.py]: I'd think we may need to use `force_text()` -- won't the next line fail if a lazy instance isn't resolved?

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by timgraham** [comment]
buildbot, add to whitelist.

File: django/db/models/fields/__init__.py (hunk 1 above)
Comment (Pradeek) [inline django/db/models/fields/__init__.py]: True. Should've checked for that. Will update it.

**Comment by Pradeek** [comment]
@timgraham Can you checkout the CI errors? Looks like its unable to checkout the rev?

**Comment by timgraham** [comment]
Could you add a test for the lazy case, squash commits, and follow the commit message guidelines as noted in our [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist)?

File: django/db/models/fields/__init__.py [hunk 2]
```diff
@@ -1963,6 +1963,8 @@ def get_internal_type(self):
         return "GenericIPAddressField"
 
     def to_python(self, value):
+        if value and not isinstance(value, six.string_types):
```
Comment (kezabelle) [inline django/db/models/fields/__init__.py]: You've lost the `value = value.strip()` (line 1223 as of `master` at 8714403614c4dfa37e806db4a6708b8a91a827a4)

Comment (Pradeek) [inline django/db/models/fields/__init__.py]: Good catch. Fixed it.

**Comment by Pradeek** [comment]
@timgraham Can you check the CI errors again? I've added the relevant tests, squashed the commits and updated the PR. The tests were passing just before squashing. Now it says `FATAL: Could not checkout 2e0563dfd9130ca77fc41babbad66e54026374ca` Any suggestions?

**Comment by timgraham** [comment]
Sometimes those checkout errors happen from time to time, no need to worry about them.

File: django/db/models/fields/__init__.py (hunk 2 above)
Comment (timgraham) [inline django/db/models/fields/__init__.py]: Could we add something like:
```
if value is None:
    return None
```
to avoid checking `if value:` 3 times?

**Comment by timgraham** [comment]
'Fixes' -> 'Fixed' in the commit message; otherwise, this looks good.

**Comment by Pradeek** [comment]
@timgraham Done.

**Comment by timgraham** [comment]
merged in 6123e6134fba627a49ad379edfc20ae0f73bc9ac, thanks!

**Pull request closed without merging**